packages/xml-views-completion/src/api.ts

```diff
@@ -36,7 +37,36 @@ export function getXMLViewCompletions(
       includes(allowedVisibility, _.ui5Node.visibility)
   );
 
-  return publicAndProtectedSuggestions;
+  const filteredSuggestions = filterBySettings(
+    publicAndProtectedSuggestions,
+    opts.settings
+  );
+
+  return filteredSuggestions;
+}
+
+function filterBySettings(
+  suggestions: UI5XMLViewCompletion[],
+  settings: CodeAssistSettings
+): UI5XMLViewCompletion[] {
+  const filteredSuggestions = suggestions;
```

Comment:
If `remove` is better in this function (vs filter/reject), then you can also shallow clone the suggestions before mutating them.

>  const filteredSuggestions = clone(suggestions);